lib/modules/manager/gitlabci-include/extract.ts

```diff
@@ -68,6 +79,17 @@ function extractDepFromIncludeComponent(
     );
     dep.skipReason = 'unsupported-version';
   }
+  const endpointUrl = parseUrl(endpoint);
+  if (
+    endpointUrl &&
+    endpointUrl.hostname !== componentReferenceMatch.groups.fqdn
+  ) {
+    logger.debug(
+      { componentReference: includeComponent.component },
+      'Ignoring external component reference',
+    );
+    dep.skipReason = 'invalid-value';
```

Comment:
why?

Reply:
Because that's what [GitLab says](https://docs.gitlab.com/ee/ci/components/#use-a-component) :)

> gitlab.example.com is the Fully Qualified Domain Name (FQDN) matching the GitLab host. You can only reference components in the same GitLab instance as your project.

Reply:
we should simply try to update those too, maybe in a future version gitlab supports that. so we are prepared for that but trying to use this now will fail on gitlab pipeline but not on renovate.

Reply:
Hm, I do not agree here. I would be rather confused if Renovate creates MRs that are known to be broken. I see your argument but there is no indication that GitLab will support external references in the near future. Maybe we can create a follow-up issue to recheck this at a later point?

Reply:
the pipeline is already broken if you use external references, so remove this check or allow external references.

Reply:
Good point 😀 I'll try to support them then.